reg_tests/chgres_cube/driver.hercules.sh

```diff
@@ -0,0 +1,238 @@
+
+set -x
+
+source ../../sorc/machine-setup.sh > /dev/null 2>&1
+module use ../../modulefiles
+module load build.$target.intel
+module list
+
+ulimit -s unlimited
+
+export OUTDIR="${WORK_DIR:-/work/noaa/stmp/$LOGNAME}"
+export OUTDIR="${OUTDIR}/reg-tests/chgres-cube"
+
+PROJECT_CODE="${PROJECT_CODE:-nesdis-rdo2}"
+QUEUE="${QUEUE:-batch}"
+
+#-----------------------------------------------------------------------------
+# Should not have to change anything below here.  HOMEufs is the root
+# directory of your UFS_UTILS clone.  HOMEreg contains the input data
+# and baseline data for each test.
+#-----------------------------------------------------------------------------
+
+export UPDATE_BASELINE="FALSE"
+#export UPDATE_BASELINE="TRUE"
+
+if [ "$UPDATE_BASELINE" = "TRUE" ]; then
+  source ../get_hash.sh
+fi
+
+export HOMEufs=$PWD/../..
+
+# TODO change this to an official location when testing is complete.
+#export HOMEreg=/work/noaa/nems/role-nems/hercules/ufs_utils/reg_tests/chgres_cube
+export HOMEreg=/work/noaa/global/dhuber/noscrub/ufs_utils/reg_tests/chgres_cube
+
+LOG_FILE=consistency.log
+SUM_FILE=summary.log
+rm -f $SUM_FILE ${LOG_FILE}*
+
+export OMP_STACKSIZE=1024M
+
+export APRUN=srun
+
+export machine=hercules
+
+export NCCMP=${NCCMP:-nccmp}
+
+rm -fr $OUTDIR
+
+#-----------------------------------------------------------------------------
+# Initialize C96 using FV3 warm restart files.
+#-----------------------------------------------------------------------------
+
+LOG_FILE1=${LOG_FILE}01
+export OMP_NUM_THREADS=1  # needs to match cpus-per-task
+TEST1=$(sbatch --parsable --ntasks-per-node=6 --nodes=1 --mem=75G -t 0:15:00 -A $PROJECT_CODE -q $QUEUE -J c96.fv3.restart \
+      -o $LOG_FILE1 -e $LOG_FILE1 ./c96.fv3.restart.sh)
+
+#-----------------------------------------------------------------------------
+# Initialize C192 using FV3 tiled history files.
+#-----------------------------------------------------------------------------
+
+LOG_FILE2=${LOG_FILE}02
+export OMP_NUM_THREADS=1  # needs to match cpus-per-task
+TEST2=$(sbatch --parsable --ntasks-per-node=6 --nodes=1 --mem=75G -t 0:15:00 -A $PROJECT_CODE -q $QUEUE -J c192.fv3.history \
+      --open-mode=append -o $LOG_FILE2 -e $LOG_FILE2 ./c192.fv3.history.sh)
+
+#-----------------------------------------------------------------------------
+# Initialize C96 using FV3 gaussian nemsio files.
+#-----------------------------------------------------------------------------
+
+LOG_FILE3=${LOG_FILE}03
+export OMP_NUM_THREADS=1  # needs to match cpus-per-task
+TEST3=$(sbatch --parsable --ntasks-per-node=6 --nodes=1 --mem=75G -t 0:15:00 -A $PROJECT_CODE -q $QUEUE -J c96.fv3.nemsio \
+      --open-mode=append -o $LOG_FILE3 -e $LOG_FILE3 ./c96.fv3.nemsio.sh)
+
+#-----------------------------------------------------------------------------
+# Initialize C96 using spectral GFS sigio/sfcio files.
+#-----------------------------------------------------------------------------
+
```

Comment:
This test almost timed out for me. Can you boost the wall clock time by 5 minutes? 

This test runs 5 minutes longer on Hercules compared to Orion. The slow down is in the read of the input atmospheric file. That file contains spectral coefficients that are converted to grid point space using the SP library. Using threads speeds up this conversion. In my experience, when this test runs slow it is because the SP library was not compiled with threads. May want to alert the libraries team.

Reply:
Noted, thanks George.

@AlexanderRichert-NOAA would you be able to verify if the SP library was built with threading in spack-stack/1.5.0 on Hercules?

Reply:
FYI Hercules is offline today for maintenance and I am off tomorrow.  I will return to this on Monday.

Reply:
Sorry I missed this-- judging by the output at /work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-1.5.0/envs/unified-env/install/intel/2021.9.0/sp-2.3.3-f4hmlri/.spack/spack-build-out.txt, it appears it was built with openmp support (-DOPENMP=ON, and the -DOPENMP definition is in the compiler calls).